### ISSUE TITLE
Pending items on Course View page

### ISSUE THREAD

**Issue by ishanivardhan** [NHSLeadership/nightingale-moodle#44]
As per original designs, it would be nice to have an entire Course Index listing on the right of every course activity view page but this needs quite a bit of work & time and hence will be pursued as a separate ticket towards the end if time permits OR in the next version

See below design for reference:
![Original Course View design](https://user-images.githubusercontent.com/25176815/32226408-70891a4a-be40-11e7-9c00-f6d5918092ae.png)

cc: @cehwitham , @Pro-Paul

**Comment by cehwitham** [comment]
Thank you for the update @ishanivardhan. Could you share some of the complexities around why this will take lots of work please? I think as you say, it's sensible if we don't let this hold up the v1.0 release.

**Comment by ishanivardhan** [comment]
Yes @cehwitham , 
- Basically, the current course index listing logic cannot be picked up as is to serve the design. Possibly a separate function needs to be built that only has the bits we need as per the design and this could be tricky given the logic sits in various course renderer & course format renderer functions that needs to be extended or invoked in the right manner.

- Style wise as well, as earlier tried in Course listing ticket, it's not very straightforward to make the sections expand-collapse the way it's in the design. Simply applying Nightingale classes as they are in header sub-menus doesn't make it work. So, need to check as to what combination of JS & PHP logic will we need once we have the markup in a separate function as mentioned in pt[1]. 

All of this needs to nicely be brought together that requires more time and a bit of trial-n-error first.

**Comment by penpadstudio** [comment]
As stated in requested review my feedback:

![32232217-79e2e67a-be4f-11e7-922e-8d654d8611a1](https://user-images.githubusercontent.com/23170050/32239022-47b36f78-be61-11e7-9638-bc19bdd5a7dc.png)

**Better to see if we can swap complete button for checkbox when we can**

![32232234-81a9db02-be4f-11e7-8e9c-7d61523d643d](https://user-images.githubusercontent.com/23170050/32239069-7271dd62-be61-11e7-9eb7-ecabbadfc946.png)

**Why is title blue? Should be black even if link. Again update checkbox styling when appropriate time or when you have discussed further with Harry. Also I will look into theory of Green button/Submit rule for pages. Possibly add spacing between buttons for short term goal.**

![32232237-823cde16-be4f-11e7-927f-84599a94f6c9](https://user-images.githubusercontent.com/23170050/32239253-f5960790-be61-11e7-8974-c15691d4ccb9.png)

**I will look into activities layouts for more refinement and iteration further down the line.**

**Comment by ishanivardhan** [comment]
As per feedback from @adamref , few things he suggested that could improvise Participant flow/experience:

1. **Mark complete** & **Mark incomplete** buttons are both in green with just text change. Could do with background color change for one of them.
@Pro-Paul - your input required. A change to make it checkbox is not advisable for now as this is part of a plugin, not theme. But you could suggest button color change that can be done via CSS.

2. No easy way for Student to have overview of the entire unit once they are on an activity page. So, maybe a course index on left or right of each activity page.
This was already raised n this issue above and part of original Paul’s design, so this can be given another shot now as other big tickets are over!

3. The sections on right of Course listing page to have a border around them, be more boxed.
@Pro-Paul - Your thoughts?

cc: @cehwitham

**Comment by penpadstudio** [comment]
If we change the buttons to the following visual format (similar to forums):
![screen shot 2017-11-27 at 14 46 46](https://user-images.githubusercontent.com/23170050/33272369-de95369e-d381-11e7-82f1-6a6e9a9a87d4.png)

Then we can possibly keep the completion button in green above and aligned right temporarily.

Also thanks @adamref for the input on boxing the modules, maybe we can implement this in a more styled approach next version.

**Comment by ishanivardhan** [comment]
@jdbrew , @cehwitham - Need your view on whether or not we should have course listing index on every course element page?

**Comment by jdbrew** [comment]
@ishanivardhan Let's research it for post v1.

Seacole 1 had it and it wasn't a great experience and impacted load times and accessibility. No programme since has had it. We've tried to stick to the GDS principle of one thing per page. If we want to add it back in let's show that the new design evaluates well against needs without impacting the existing page needs.

**Comment by ishanivardhan** [comment]
Fair enough! Thanks for your input @jdbrew .. I'll move this in Icebox and come back to it when we discuss post v1 sprint planning to take up on priority.
